src/puppetlabs/puppetdb/queue.clj

```diff
@@ -190,47 +198,65 @@
    :certname s/Str
    :producer-ts (s/maybe pls/Timestamp)
    :callback (s/=> s/Any s/Any)
-   :command-stream java.io.InputStream})
+   :command-stream java.io.InputStream
+   :compression s/Str})
 
 (s/defn create-command-req :- command-req-schema
   "Validating constructor function for command requests"
   [command :- s/Str
    version :- s/Int
    certname :- s/Str
    producer-ts :- (s/maybe s/Str)
+   compression :- s/Str
    callback :- (s/=> s/Any s/Any)
    command-stream :- java.io.InputStream]
   {:command command
    :version version
    :certname certname
    :producer-ts (when producer-ts
                   (pdbtime/from-string producer-ts))
+   :compression compression
    :callback callback
    :command-stream command-stream})
 
-(defrecord CommandRef [id command version certname received producer-ts callback delete?])
+(defrecord CommandRef [id command version certname received producer-ts callback delete? compression])
 
 (defn cmdref->entry [{:keys [id received] :as cmdref}]
   (stock/entry id (serialize-metadata received cmdref)))
 
 (defn entry->cmdref [entry]
-  (let [{:keys [received command version certname producer-ts]} (-> entry
-                                                                    stock/entry-meta
-                                                                    parse-metadata)]
+  (let [{:keys [received command version
+                certname producer-ts compression]} (-> entry
+                                                       stock/entry-meta
+                                                       parse-metadata)]
     (map->CommandRef {:id (stock/entry-id entry)
                       :command command
                       :version version
                       :certname certname
                       :received received
                       :producer-ts producer-ts
-                      :callback identity})))
+                      :callback identity
+                      :compression compression})))
+
+(defn wrap-decompression-stream
+  [compression command-stream]
```

Comment:
Sure, so just rename `compression` to `file-extension` in this one function, then - or were you thinking I should do that in some other functions as well?